metaflow/plugins/kubernetes/kubernetes_job.py

```diff
@@ -71,6 +71,13 @@ def create(self):
         # Note: This implementation ensures that there is only one unique Pod
         # (unique UID) per Metaflow task attempt.
         client = self._client.get()
+        tolerations = []
+        for toleration in self._kwargs.get("tolerations") or []:
```

Comment:
to maintain the same coding style in this file, would you mind in-lining this in L:198 instead?

Reply:
I extracted the code to handle errors if the input is invalid. I can't do it in-line.
I can extract it in a function, WDYT?

Reply:
we can move the input verification inside the `@kubernetes` decorator module - currently all validations (we don't have many) reside there. Also, the `@kubernetes` decorator hooks are guaranteed to execute for both `@kubernetes` and `argo-workflows create` while `kubernetes_job` only gets invoked for `python flow.py run` - so this toleration check would not happen for argo workflows.

Reply:
in `KubernetesDecorator`, `KUBERNETES_TOLERATIONS` is a string and not a parsed JSON.
I can try to move the logic in `Kubernetes.create_job()`, WDYT?

Reply:
The problem is this piece of code in `KubernetesDecorator`
```python
if not self.attributes["node_selector"] and KUBERNETES_NODE_SELECTOR:
    self.attributes["node_selector"] = KUBERNETES_NODE_SELECTOR
```
`self.attributes["node_selector"]` is a python object set as a parameter to the `@kubernetes` decorator, but `KUBERNETES_NODE_SELECTOR` is a string set using an environment variable.
`KUBERNETES_NODE_SELECTOR` is parsed as a python object in `kubernetes_cli`, which is executed afterward.

One solution would be to parse `KUBERNETES_NODE_SELECTOR` in `KubernetesDecorator` but it would duplicate the logic done by `@click.option`

Reply:
Fixed